src/main/java/org/kohsuke/github/GHRepository.java

```diff
@@ -810,6 +851,18 @@ public GHRef getRef(String refName) throws IOException {
         refName = refName.replaceAll("#", "%23");
         return root.retrieve().to(String.format("/repos/%s/%s/git/refs/%s", getOwnerName(), name, refName), GHRef.class).wrap(root);
     }
+
+    /**
+     * Returns the <strong>annotated</strong> tag object. Only valid if the {@link GHRef#getObject()} has a
+     * {@link GHRef.GHObject#getType()} of {@code tag}.
+     *
+     * @param sha the sha of the tag object
+     * @return the annotated tag object
+     */
+    public GHTagObject getTagObject(String sha) throws IOException {
+        return root.retrieve().to(getApiTailUrl("git/tags/" + sha), GHTagObject.class).wrap(this);
```

Comment:
is really only sha?

Reply:
Nope it's the tagged and the tag date and the sha of the tagged commit.

For an annotated tag object the tag ref points to a tag object that points to the commit

For a lightweight tag, the tag ref just points to the commit.

Reply:
S/tagged/tagger/